### PR TITLE
CRD Update

### PR DESCRIPTION
**Pull request by vaibhavmew**
Signed-off-by: Vaibhav Mewada <vaibhav@accuknox.com>

### CODE REVIEW

**Comment by nyrahul** [comment]
Hey @vaibhavmew , a quick grep on external reveals many other places where external-workload keyword is used ... Can we handle as much as possible in this PR? @seswarrajan, WDYT?

```
❯ grep externalworkload . -ri
./helm/templates/security.kubearmor.com_kubearmorexternalworkloads.yaml:  name: kubearmorexternalworkloads.security.kubearmor.com
./helm/templates/security.kubearmor.com_kubearmorexternalworkloads.yaml:    kind: KubeArmorExternalWorkload
./helm/templates/security.kubearmor.com_kubearmorexternalworkloads.yaml:    listKind: KubeArmorExternalWorkloadList
./helm/templates/security.kubearmor.com_kubearmorexternalworkloads.yaml:    plural: kubearmorexternalworkloads
./helm/templates/security.kubearmor.com_kubearmorexternalworkloads.yaml:    singular: kubearmorexternalworkload
./helm/templates/security.kubearmor.com_kubearmorexternalworkload.yaml:  name: kubearmorexternalworkloads.security.kubearmor.com
./helm/templates/security.kubearmor.com_kubearmorexternalworkload.yaml:    kind: KubeArmorExternalWorkload
./helm/templates/security.kubearmor.com_kubearmorexternalworkload.yaml:    listKind: KubeArmorExternalWorkloadList
./helm/templates/security.kubearmor.com_kubearmorexternalworkload.yaml:    plural: kubearmorexternalworkloads
./helm/templates/security.kubearmor.com_kubearmorexternalworkload.yaml:    singular: kubearmorexternalworkload
./deployments/CRD/KubeArmorExternalWorkloadPolicy.yaml:  name: kubearmorexternalworkloads.security.kubearmor.com
./deployments/CRD/KubeArmorExternalWorkloadPolicy.yaml:    kind: KubeArmorExternalWorkload
./deployments/CRD/KubeArmorExternalWorkloadPolicy.yaml:    listKind: KubeArmorExternalWorkloadList
./deployments/CRD/KubeArmorExternalWorkloadPolicy.yaml:    plural: kubearmorexternalworkloads
./deployments/CRD/KubeArmorExternalWorkloadPolicy.yaml:    singular: kubearmorexternalworkload
./deployments/CRD/KubeArmorExternalWorkload.yaml:  name: kubearmorexternalworkloads.security.kubearmor.com
./deployments/CRD/KubeArmorExternalWorkload.yaml:    kind: KubeArmorExternalWorkload
./deployments/CRD/KubeArmorExternalWorkload.yaml:    listKind: KubeArmorExternalWorkloadList
./deployments/CRD/KubeArmorExternalWorkload.yaml:    plural: kubearmorexternalworkloads
./deployments/CRD/KubeArmorExternalWorkload.yaml:    singular: kubearmorexternalworkload
./src/types/types.go:type K8sKubeArmorExternalWorkloadPolicyStatus struct {
./src/types/types.go:// K8sKubeArmorExternalWorkloadPolicyEvent Structure
./src/types/types.go:type K8sKubeArmorExternalWorkloadPolicyEvent struct {
./src/types/types.go:	Object K8sKubeArmorExternalWorkloadPolicy `json:"object"`
./src/types/types.go:// K8sKubeArmorExternalWorkloadPolicy Structure
./src/types/types.go:type K8sKubeArmorExternalWorkloadPolicy struct {
./src/types/types.go:	Spec     ExternalWorkloadSecuritySpec             `json:"spec"`
./src/types/types.go:	Status   K8sKubeArmorExternalWorkloadPolicyStatus `json:"status,omitempty"`
./src/types/types.go:// K8sKubeArmorExternalWorkloadPolicies Structure
./src/types/types.go:type K8sKubeArmorExternalWorkloadPolicies struct {
./src/types/types.go:	Items []K8sKubeArmorExternalWorkloadPolicy `json:"items"`
./src/types/types.go:// ExternalWorkloadSecuritySpec Structure
./src/types/types.go:type ExternalWorkloadSecuritySpec struct {
./src/types/types.go:type ExternalWorkloadMetadata struct {
./src/types/types.go:// ExternalWorkloadSecurityPolicy Structure
./src/types/types.go:type ExternalWorkloadSecurityPolicy struct {
./src/types/types.go:	Metadata ExternalWorkloadMetadata            `json:"metadata"`
./src/operator/Makefile:	cd $(CURDIR); sudo -E ./kvms-operator #-logPath=/tmp/kvms-operator.log -enableHostPolicy -enableExternalWorkloadPolicy
./src/operator/core/kvmOperator.go:	EnableExternalWorkloadPolicy bool
./src/operator/core/kvmOperator.go:	ExternalWorkloadSecurityPolicies     []tp.ExternalWorkloadSecurityPolicy
./src/operator/core/kvmOperator.go:	ExternalWorkloadSecurityPoliciesLock *sync.RWMutex
./src/operator/core/kvmOperator.go:	MapExternalWorkloadConnIdentity map[uint16]ClientConn
./src/operator/core/kvmOperator.go:	dm.ExternalWorkloadSecurityPoliciesLock = new(sync.RWMutex)
./src/operator/core/kvmOperator.go:	dm.MapExternalWorkloadConnIdentity = make(map[uint16]ClientConn)
./src/operator/core/kvmOperator.go:		go dm.WatchExternalWorkloadSecurityPolicies()
./src/operator/core/crdHandler.go:// UpdateExternalWorkloadSecurityPolicies Function
./src/operator/core/crdHandler.go:func (dm *KVMSOperator) UpdateExternalWorkloadSecurityPolicies() {
./src/operator/core/crdHandler.go:	dm.ExternalWorkloadSecurityPoliciesLock.Lock()
./src/operator/core/crdHandler.go:	defer dm.ExternalWorkloadSecurityPoliciesLock.Unlock()
./src/operator/core/crdHandler.go:	secPolicies := []tp.ExternalWorkloadSecurityPolicy{}
./src/operator/core/crdHandler.go:	for _, policy := range dm.ExternalWorkloadSecurityPolicies {
./src/operator/core/crdHandler.go:func (dm *KVMSOperator) GenerateExternalWorkloadIdentity(name string, labels map[string]string) uint16 {
./src/operator/core/crdHandler.go:func (dm *KVMSOperator) GetExternalWorkloadIdentities(name string) []uint16 {
./src/operator/core/crdHandler.go:func (dm *KVMSOperator) GetExternalWorkloadLabel(identity uint16) string {
./src/operator/core/crdHandler.go:func (dm *KVMSOperator) GetExternalWorkLoadAllLabels() []string {
./src/operator/core/crdHandler.go:	var externalWorkloadLabels []string
./src/operator/core/crdHandler.go:		externalWorkloadLabels = append(externalWorkloadLabels, label)
./src/operator/core/crdHandler.go:	return externalWorkloadLabels
./src/operator/core/crdHandler.go:// WatchExternalWorkloadSecurityPolicies Function
./src/operator/core/crdHandler.go:func (dm *KVMSOperator) WatchExternalWorkloadSecurityPolicies() {
./src/operator/core/crdHandler.go:		if resp := K8s.WatchK8sExternalWorkloadSecurityPolicies(); resp != nil {
./src/operator/core/crdHandler.go:				event := tp.K8sKubeArmorExternalWorkloadPolicyEvent{}
./src/operator/core/crdHandler.go:				dm.ExternalWorkloadSecurityPoliciesLock.Lock()
./src/operator/core/crdHandler.go:				secPolicy := tp.ExternalWorkloadSecurityPolicy{}
./src/operator/core/crdHandler.go:					if !kl.ContainsElement(dm.ExternalWorkloadSecurityPolicies, secPolicy) {
./src/operator/core/crdHandler.go:						dm.ExternalWorkloadSecurityPolicies = append(dm.ExternalWorkloadSecurityPolicies, secPolicy)
./src/operator/core/crdHandler.go:						identity := dm.GenerateExternalWorkloadIdentity(secPolicy.Metadata.Name, secPolicy.Metadata.NodeSelector.MatchLabels)
./src/operator/core/crdHandler.go:					for idx, policy := range dm.ExternalWorkloadSecurityPolicies {
./src/operator/core/crdHandler.go:							dm.ExternalWorkloadSecurityPolicies[idx] = secPolicy
./src/operator/core/crdHandler.go:					for idx, policy := range dm.ExternalWorkloadSecurityPolicies {
./src/operator/core/crdHandler.go:							dm.ExternalWorkloadSecurityPolicies = append(dm.ExternalWorkloadSecurityPolicies[:idx], dm.ExternalWorkloadSecurityPolicies[idx+1:]...)
./src/operator/core/crdHandler.go:				dm.ExternalWorkloadSecurityPoliciesLock.Unlock()
./src/operator/core/crdHandler.go:				dm.UpdateExternalWorkloadSecurityPolicies()
./src/operator/core/k8sHandler.go:// WatchK8sExternalWorkloadSecurityPolicies Function
./src/operator/core/k8sHandler.go:func (kh *K8sHandler) WatchK8sExternalWorkloadSecurityPolicies() *http.Response {
./src/operator/core/k8sHandler.go:		URL := "https://" + kh.K8sHost + ":" + kh.K8sPort + "/apis/security.kubearmor.com/v1/kubearmorexternalworkloads?watch=true"
./src/operator/core/k8sHandler.go:	URL := "http://" + kh.K8sHost + ":" + kh.K8sPort + "/apis/security.kubearmor.com/v1/kubearmorexternalworkloads?watch=true"
./src/constants/constants.go:	KewCRDName         = "kubearmorexternalworkloads"
./src/service/Makefile:	cd $(CURDIR); sudo -E ./kvmservice #-logPath=/tmp/kvmservice.log -enableHostPolicy -enableExternalWorkloadPolicy
./src/service/core/kvmService.go:	ExternalWorkloadSecurityPolicies     []tp.ExternalWorkloadSecurityPolicy
./src/service/core/kvmService.go:	ExternalWorkloadSecurityPoliciesLock *sync.RWMutex
./src/service/core/kvmService.go:	MapExternalWorkloadConnIdentity map[uint16]ClientConn
./src/service/core/kvmService.go:	dm.ExternalWorkloadSecurityPoliciesLock = new(sync.RWMutex)
./src/service/core/kvmService.go:	dm.MapExternalWorkloadConnIdentity = make(map[uint16]ClientConn)
./src/service/genscript/genscript.go:func GenerateEWInstallationScript(externalWorkload, identity string) string {
./src/service/genscript/genscript.go:	kg.Printf("ClusterIP:%s ClusterPort:%d ewName:%s identity:%s", p.ipAddress, p.port, externalWorkload, identity)
./getting-started/minikube/README.md:After starting minikube, apply the VM/ExternalWorkload and Hostpolicy CRD using below commands.
./getting-started/minikube/README.md:$ minikube kubectl -- apply -f ../../deployments/CRD/KubeArmorExternalWorkloadPolicy.yaml 
./getting-started/minikube/README.md:customresourcedefinition.apiextensions.k8s.io/kubearmorexternalworkloads.security.kubearmor.com created
./getting-started/minikube/README.md:kubearmorexternalworkloadpolicy.security.kubearmor.com/external-workload-01 created
grep: ./.git/index: binary file matches
./pkg/KubeArmorExternalWorkload/config/rbac/role.yaml:  - kubearmorexternalworkload
./pkg/KubeArmorExternalWorkload/config/rbac/role.yaml:  - kubearmorexternalworkloadstatus
./pkg/KubeArmorExternalWorkload/config/crd/bases/security.kubearmor.com_kubearmorexternalworkloads.yaml:  name: kubearmorexternalworkloads.security.kubearmor.com
./pkg/KubeArmorExternalWorkload/config/crd/bases/security.kubearmor.com_kubearmorexternalworkloads.yaml:    kind: KubeArmorExternalWorkload
./pkg/KubeArmorExternalWorkload/config/crd/bases/security.kubearmor.com_kubearmorexternalworkloads.yaml:    listKind: KubeArmorExternalWorkloadList
./pkg/KubeArmorExternalWorkload/config/crd/bases/security.kubearmor.com_kubearmorexternalworkloads.yaml:    plural: kubearmorexternalworkloads
./pkg/KubeArmorExternalWorkload/config/crd/bases/security.kubearmor.com_kubearmorexternalworkloads.yaml:    singular: kubearmorexternalworkload
./pkg/KubeArmorExternalWorkload/config/crd/bases/security.kubearmor.com_kubearmorexternalworkload.yaml:  name: kubearmorexternalworkloads.security.kubearmor.com
./pkg/KubeArmorExternalWorkload/config/crd/bases/security.kubearmor.com_kubearmorexternalworkload.yaml:    kind: KubeArmorExternalWorkload
./pkg/KubeArmorExternalWorkload/config/crd/bases/security.kubearmor.com_kubearmorexternalworkload.yaml:    listKind: KubeArmorExternalWorkloadList
./pkg/KubeArmorExternalWorkload/config/crd/bases/security.kubearmor.com_kubearmorexternalworkload.yaml:    plural: kubearmorexternalworkloads
./pkg/KubeArmorExternalWorkload/config/crd/bases/security.kubearmor.com_kubearmorexternalworkload.yaml:    singular: kubearmorexternalworkload
./pkg/KubeArmorExternalWorkload/config/crd/kustomization.yaml:- bases/security.kubearmor.com_kubearmorexternalworkload.yaml
./pkg/KubeArmorExternalWorkload/config/crd/kustomization.yaml:#- patches/webhook_in_kubearmorexternalworkload.yaml
./pkg/KubeArmorExternalWorkload/config/crd/kustomization.yaml:#- patches/cainjection_in_kubearmorexternalworkload.yaml
grep: ./pkg/KubeArmorExternalWorkload/bin/manager: binary file matches
./pkg/KubeArmorExternalWorkload/PROJECT:  kind: KubeArmorExternalWorkload
./pkg/KubeArmorExternalWorkload/Makefile:#security.kubearmor.com_kubearmorexternalworkload.yaml
./pkg/KubeArmorExternalWorkload/Makefile:	cp config/crd/bases/security.kubearmor.com_kubearmorexternalworkload.yaml ../../helm/templates/security.kubearmor.com_kubearmorexternalworkload.yaml
./pkg/KubeArmorExternalWorkload/Makefile:	cp config/crd/bases/security.kubearmor.com_kubearmorexternalworkload.yaml ../../deployments/CRD/KubeArmorExternalWorkload.yaml
./pkg/KubeArmorExternalWorkload/Makefile:	kustomize build config/default > /tmp/KubeArmorExternalWorkload.yaml
./pkg/KubeArmorExternalWorkload/Makefile:	rm -rf bin go.sum cover.out /tmp/KubeArmorExternalWorkload.yaml
./pkg/KubeArmorExternalWorkload/main.go:	securityv1 "github.com/kubearmor/KubeArmor/pkg/KubeArmorExternalWorkload/api/v1"
./pkg/KubeArmorExternalWorkload/main.go:	"github.com/kubearmor/KubeArmor/pkg/KubeArmorExternalWorkload/controllers"
./pkg/KubeArmorExternalWorkload/main.go:	if err = (&controllers.KubeArmorExternalWorkloadReconciler{
./pkg/KubeArmorExternalWorkload/main.go:		Log:    ctrl.Log.WithName("controllers").WithName("KubeArmorExternalWorkload"),
./pkg/KubeArmorExternalWorkload/main.go:		setupLog.Error(err, "unable to create controller", "controller", "KubeArmorExternalWorkload")
./pkg/KubeArmorExternalWorkload/go.mod:module github.com/kubearmor/KubeArmor/pkg/KubeArmorExternalWorkload
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:func (in *KubeArmorExternalWorkload) DeepCopyInto(out *KubeArmorExternalWorkload) {
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new KubeArmorExternalWorkload.
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:func (in *KubeArmorExternalWorkload) DeepCopy() *KubeArmorExternalWorkload {
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:	out := new(KubeArmorExternalWorkload)
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:func (in *KubeArmorExternalWorkload) DeepCopyObject() runtime.Object {
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:func (in *KubeArmorExternalWorkloadList) DeepCopyInto(out *KubeArmorExternalWorkloadList) {
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:		*out = make([]KubeArmorExternalWorkload, len(*in))
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new KubeArmorExternalWorkloadList.
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:func (in *KubeArmorExternalWorkloadList) DeepCopy() *KubeArmorExternalWorkloadList {
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:	out := new(KubeArmorExternalWorkloadList)
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:func (in *KubeArmorExternalWorkloadList) DeepCopyObject() runtime.Object {
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:func (in *KubeArmorExternalWorkloadStatus) DeepCopyInto(out *KubeArmorExternalWorkloadStatus) {
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new KubeArmorExternalWorkloadStatus.
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:func (in *KubeArmorExternalWorkloadStatus) DeepCopy() *KubeArmorExternalWorkloadStatus {
./pkg/KubeArmorExternalWorkload/api/v1/zz_generated.deepcopy.go:	out := new(KubeArmorExternalWorkloadStatus)
./pkg/KubeArmorExternalWorkload/api/v1/kubearmorexternalworkload_types.go:// +kubebuilder:resource:singular="kubearmorexternalworkload",path="kubearmorexternalworkloads",scope="Cluster",shortName={kew}
./pkg/KubeArmorExternalWorkload/api/v1/kubearmorexternalworkload_types.go:type KubeArmorExternalWorkload struct {
./pkg/KubeArmorExternalWorkload/api/v1/kubearmorexternalworkload_types.go:	Status KubeArmorExternalWorkloadStatus `json:"status"`
./pkg/KubeArmorExternalWorkload/api/v1/kubearmorexternalworkload_types.go:// KubeArmorExternalWorkloadStatus is the status of a the external KubeArmor workload.
./pkg/KubeArmorExternalWorkload/api/v1/kubearmorexternalworkload_types.go:type KubeArmorExternalWorkloadStatus struct {
./pkg/KubeArmorExternalWorkload/api/v1/kubearmorexternalworkload_types.go:// KubeArmorExternalWorkloadList is a list of KubeArmorExternalWorkload objects.
./pkg/KubeArmorExternalWorkload/api/v1/kubearmorexternalworkload_types.go:type KubeArmorExternalWorkloadList struct {
./pkg/KubeArmorExternalWorkload/api/v1/kubearmorexternalworkload_types.go:	// Items is a list of KubeArmorExternalWorkload
./pkg/KubeArmorExternalWorkload/api/v1/kubearmorexternalworkload_types.go:	Items []KubeArmorExternalWorkload `json:"items"`
./pkg/KubeArmorExternalWorkload/api/v1/kubearmorexternalworkload_types.go:	SchemeBuilder.Register(&KubeArmorExternalWorkload{}, &KubeArmorExternalWorkloadList{})
./pkg/KubeArmorExternalWorkload/controllers/suite_test.go:	securityv1 "github.com/kubearmor/KubeArmor/pkg/KubeArmorExternalWorkload/api/v1"
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:	securityv1 "github.com/kubearmor/KubeArmor/pkg/KubeArmorExternalWorkload/api/v1"
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:// KubeArmorExternalWorkloadReconciler reconciles a KubeArmorExternalWorkload object
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:type KubeArmorExternalWorkloadReconciler struct {
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:// +kubebuilder:rbac:groups=security.kubearmor.com,resources=kubearmorexternalworkload,verbs=get;list;watch;create;update;patch;delete
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:// +kubebuilder:rbac:groups=security.kubearmor.com,resources=kubearmorexternalworkloadstatus,verbs=get;update;patch
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:func (r *KubeArmorExternalWorkloadReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:	log := r.Log.WithValues("kubearmorexternalworkload", req.NamespacedName)
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:	policy := &securityv1.KubeArmorExternalWorkload{}
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:		log.Info("Invalid KubeArmorExternalWorkload")
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:	// Validate KubeArmorExternalWorkload
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:	log.Info("Fetched KubeArmorExternalWorkload")
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:func (r *KubeArmorExternalWorkloadReconciler) SetupWithManager(mgr ctrl.Manager) error {
./pkg/KubeArmorExternalWorkload/controllers/kubearmorexternalworkload_controller.go:		For(&securityv1.KubeArmorExternalWorkload{}).
./examples/kew2policy.yaml:kind: KubeArmorExternalWorkloadPolicy
./examples/kewpolicy.yaml:kind: KubeArmorExternalWorkload
```

**Comment by vaibhavmew** [comment]
@nyrahul Few more files need changes. I will push them immediately.

File: src/constants/constants.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ var (
 	KvmOprIdentityToLabel   = "/kvm-opr-identity-to-label-maps/"
 
 	KhpCRDName         = "kubearmorhostpolicies"
-	KewCRDName         = "kubearmorexternalworkloads"
+	KewCRDName         = "kubearmorvirtualmachines"
```
Comment (seswarrajan) [inline src/constants/constants.go:13]: Also kindly modify all kew references to kvm.

**Review by seswarrajan — COMMENTED** [review]
As also mentioned by @nyrahul, I also noted that there are multiple references to workload. Can you check all once.

**Review by nyrahul — APPROVED** [review]
LGTM, ok to merge when @seswarrajan approves it.

**Review by seswarrajan — APPROVED** [review]

**Pull request merged**